authority/tls.go

```diff
@@ -251,6 +252,52 @@ func (a *Authority) Renew(ocx *x509.Certificate) (*x509.Certificate, *x509.Certi
 	return serverCert, caCert, nil
 }
 
+// RevocationTypeSelection is a map representing the selected types of
+type RevocationTypeSelection struct {
+	All, Passive, CRL, OCSP bool
+}
+
+// Revoke revokes a certificate.
+//
+// NOTE: Only supports passive revocation - prevent existing certificates from
+// being renewed.
+//
+// TODO: Add OCSP and CRL support.
+func (a *Authority) Revoke(rts RevocationTypeSelection, serial string, provisionerID string, reason int) error {
+	var errContext = context{
+		"serial":        serial,
+		"provisionerID": provisionerID,
+		"reason":        reason,
+	}
+	if rts.All || rts.Passive {
+		if a.db == nil {
+			return &apiError{errors.New("no persistence layer configured"),
+				http.StatusUnauthorized, errContext}
+		}
+
+		rci := &db.RevokedCertificateInfo{
+			Serial:        serial,
+			ProvisionerID: provisionerID,
+			Reason:        reason,
+			RevokedAt:     time.Now().UTC(),
+		}
```

Comment:
We should store the reason code too, it's "easier" to process than a string.

Reply:
this is already the code. you also want to store the string? I'll make it more clear with the variable name that this is a code.

Reply:
No, i thought it was the string.